konnectivity-client/pkg/client/client.go

```diff
@@ -43,17 +43,18 @@ type dialResult struct {
 }
 
 // grpcTunnel implements Tunnel
-type grpcTunnel struct {
+type singleUseGrpcTunnel struct {
```

Comment:
Can we keep the grpcTunnel, and make a new struct singleUseGRPCTunnel? Btw, the singleUseGRPCTunnel doesn't need maps for connections and pending Dials.

Reply:
Renamed this back to grpcTunnel. I think it's fine to just use this struct instead of creating a new one since the goal is to eventually replace this with connection multiplexing.